susemanager-utils/mgr-libmod/mgrlibmod/mllib.py

```diff
@@ -0,0 +1,330 @@
+        if name not in MLLibmodProc.RESERVED_STREAMS:
+            self._enabled_stream_modules.pop(name, None)
+
+    def get_module_streams(self, name: str) -> List:
+        if self._mod_index is None:
+            self.index_modules()
+        assert self._mod_index is not None, "Unable to get module streams: module index not found"
+
+        streams: Set = set()
+        module = self._mod_index.get_module(name)
+        if module:
+            for s_obj in module.get_all_streams():
+                streams.add(s_obj.get_stream_name())
+
+        return list(streams)
+
+    def get_stream_contexts(self, s_type: mltypes.MLStreamType) -> List:
+        if self._mod_index is None:
+            self.index_modules()
+        assert self._mod_index is not None, "Unable to get stream contexts: module index not found"
+        contexts: List = []
+        module = self._mod_index.get_module(s_type.name)
+        if module:
+            for stream in module.get_all_streams():
+                if stream.get_stream_name() == s_type.stream:
+                    contexts.append(stream)
+
+        return contexts
+
+    def get_stream_dependencies(self, ctx: Modulemd.ModuleStreamV2) -> List[str]:
+        """
+        get_stream_dependencies -- get stream dependencies.
+
+        :param ctx: module stream context
+        :type ctx: Modulemd.ModuleStreamV2
+        :return: list of dependency names.
+        :rtype: List[str]
+        """
+        deps: List[str] = []
+        s_deps: List[Modulemd.Dependencies] = ctx.get_dependencies() or []
+        dep: Modulemd.Dependencies
+        for dep in s_deps:
+            deps.extend(dep.get_runtime_modules())
+
+        return deps
+
+    def index_modules(self) -> None:
+        """
+        index_modules -- loads given metadata and indexes modules from there.
+        """
+        if self._mod_index is None:
+            mgr: Modulemd.ModuleIndex = Modulemd.ModuleIndexMerger.new()
+            for path in self.metadata:
+                idx = Modulemd.ModuleIndex.new()
+                if self._is_meta_compressed(path):
+                    with gzip.open(path) as gzmeta:
+                        idx.update_from_string(gzmeta.read().decode("utf-8"), True)
+                else:
+                    idx.update_from_file(path, True)
+                mgr.associate_index(idx, 0)
+            self._mod_index = mgr.resolve()
+
+    def get_default_stream(self, name: str):
+        assert self._mod_index is not None, "Unable to access module index when resolving default stream"
+        module = self._mod_index.get_module(name)
+        if not module:
+            raise ValueError("Module {} not found".format(name))
+        defaults = module.get_detaults()
+        if defaults:
+            return defaults.get_default_stream()
+
+        return module.get_all_streams()[0].get_stream_name()
+
+    def get_dep_streams(self, s_obj):
+        dep = s_obj.get_dependencies()[0]  # XXX: Why just always first?
```

Comment:
`get_dependencies()` returns a `Modulemd.Dependencies` object wrapped in a list, rather than returning a list of dependencies directly. And this list is always a 1 element array. But I guess it wouldn't hurt to foolproof this by looping to operate on the whole array (there's no info on this in the API docs)

Reply:
Yeah, I already figured that out, actually. And yes, I am looping it over, because who knows. Just need to push. :-)